### PR TITLE
Enable custom test discovery by disabling app domains

### PR DESCRIPTION
**Pull request by JunTaoLuo**
#381 
Verified:
- custom test discovery in Performance Benchmarks test will work
- xunit.runner.json will not be required 
- Security repo tests all pass

cc @victorhurdugaci

### CODE REVIEW

File: KoreBuild-dotnet/build/_xunit-test.shade [hunk 1]
```diff
@@ -51,6 +51,9 @@ projectFile=''
                         testArgs = " -parallel none";
                     }
 
+                    // Removing app domains to allow custom test discovery
+                    testArgs += " -noappdomain";
```
Comment (pranavkm) [inline KoreBuild-dotnet/build/_xunit-test.shade]: This affects binding redirects (they would no longer work). What's the reason for this change?

Comment (JunTaoLuo) [inline KoreBuild-dotnet/build/_xunit-test.shade]: I have been trying to come up with a scenario with binding redirect that will be broken by this change and so far it seems to work. I have also checked a few repos, such as Security, to see if this change will break them.

Comment (natemcmaster) [inline KoreBuild-dotnet/build/_xunit-test.shade]: Last I checked, this causes most 99 of 109 tests to fail in ["Microsoft.EntityFrameworkCore.Commands.FunctionalTests"](https://github.com/aspnet/EntityFramework/tree/dev/test/Microsoft.EntityFrameworkCore.Commands.FunctionalTests)

Comment (pranavkm) [inline KoreBuild-dotnet/build/_xunit-test.shade]: Mvc has plenty of these. Try referencing a test which references two different versions of Newtonsoft.Json for e.g.

**Comment by JunTaoLuo** [comment]
Will proceed with the alternative approach of `dnx451` => `net451`.
